lib/web_ui/lib/src/engine/skwasm/skwasm_impl/shaders.dart

```diff
@@ -171,15 +206,21 @@ class SkwasmFragmentProgram implements ui.FragmentProgram {
     }
     final RuntimeEffectHandle handle = runtimeEffectCreate(sourceString);
     skStringFree(sourceString);
-    return SkwasmFragmentProgram._(name, handle);
+    return SkwasmFragmentProgram._(
+      name,
+      handle,
+      shaderData.floatCount,
+      shaderData.textureCount
+    );
   }
 
   RuntimeEffectHandle handle;
   String name;
+  int floatUniformCount;
+  int childShaderCount;
```

Comment:
Do all of the fields need to be mutable?

Reply:
Definitely not. I should change this.